### ISSUE TITLE
Shuffling wav files in dataloader does not ensure that all the training files are checked in each epoch

### ISSUE THREAD

**Issue by dkatsiros** [HarryVolek/PyTorch_Speaker_Verification#73]
As a results the model is trained on `N*M` utterances per epoch and not the whole training set. This affects the convergence as well as possible extensions of the code (e.g. early stopping).

where:
 `N=number of speakers per batch`,
 `M=number of utterances per speaker per batch` according to the referenced paper.
https://github.com/HarryVolek/PyTorch_Speaker_Verification/blob/10e159a8d3255503c0184cde4eb7097968857a31/data_load.py#L39-L40

**Comment by dkatsiros** [comment]
For _TIMIT_ dataset, where `M=9` (I think) the dataloader may be ok. The issue appears in **large** datasets such as _VoxCeleb1_ or _VoxCeleb2_ where `M>50`.

**Comment by dkatsiros** [comment]
@HarryVolek Can you check this please ? If that is the case I will pr
